src/dynafed.cpp

```diff
@@ -52,6 +52,10 @@ DynaFedParamEntry ComputeNextBlockFullCurrentParameters(const CBlockIndex* pinde
         epoch_start_height -= epoch_length;
     }
 
+    if (consensus.genesis_style == "elements" && epoch_start_height == 0 && next_height > 1 && consensus.vDeployments[Consensus::DEPLOYMENT_DYNA_FED].nStartTime == Consensus::BIP9Deployment::ALWAYS_ACTIVE) {
+        // when starting in dynafed mode, the full parameters are stored in block 1 instead of 0
+        epoch_start_height = 1;
+    }
```

Comment:
This complicated edge case makes me nervous. I am assuming I don't need to worry about it affecting liquid prod, because epoch_start_height in liquid prod can never possibly be low enough to trigger it, at this point. (Except when looking at the very start of the chain, but I assume that's been tested and works.)

It seems like there ought to be a nicer way to do this, but maybe it's too late for that.

Reply:
You don't need to worry about prod because dynafed is not "ALWAYS_ACTIVE" so even if we were regenerating a block from the first epoch somehow, it would behave the same way as always